cpp/src/arrow/compute/kernels/scalar_cast_boolean.cc

```diff
@@ -50,6 +50,7 @@ struct ParseBooleanString {
 std::vector<std::shared_ptr<CastFunction>> GetBooleanCasts() {
   auto func = std::make_shared<CastFunction>("cast_boolean", Type::BOOL);
   AddCommonCasts(Type::BOOL, boolean(), func.get());
+  AddZeroCopyCast(Type::BOOL, boolean(), boolean(), func.get());
```

Comment:
Because `AddCommonCasts` doesn't include the identity cast. Should I add the identity cast to `AddCommonCasts`?

Reply:
The question is: why do we need the identity cast for non-parametric types? Calling a no-op kernel sounds like pointless overhead.

Reply:
Ah, I think it's ok to support. I just find it a bit weird that it requires instantiating distinct kernels (is that actually the case or am I misunderstanding this?).

Reply:
It does require a distinct kernel since the cast kernels are structured so that each implements a single `from_id -> to_id` cast. Perhaps when refactoring cast we can try to make this more efficient